apps/angular/projection/src/app/ui/list-item/list-item.component.ts

```diff
@@ -1,35 +1,29 @@
-import { Component, Input } from '@angular/core';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgTemplateOutlet } from '@angular/common';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 
 @Component({
   selector: 'app-list-item',
   template: `
-    <div class="border border-grey-300 py-1 px-2 flex justify-between">
+    <ng-template #templateTest>
       {{ name }}
-      <button (click)="delete(id)">
+      <button (click)="delete()">
         <img class="h-5" src="assets/svg/trash.svg" />
       </button>
+    </ng-template>
+
+    <div class="border-grey-300 flex justify-between border px-2 py-1">
+      <ng-container *ngTemplateOutlet="templateTest"></ng-container>
```

Comment:
Question: why did you implement it like this ?

Reply:
Yes, it's a bit overengineering, it's been fixed